File: backend/models/athena.py
"""Pydantic v2 models for Athena Health API requests and responses.

These models are the contract between Juno and Athena Health.
Any change to Athena's API response shape must be reflected here first.
"""
from __future__ import annotations

from pydantic import BaseModel, ConfigDict


class AthenaTokenRequest(BaseModel):
    """POST /oauth2/v1/token"""

    model_config = ConfigDict(extra="allow")

    client_id: str
    client_secret: str
    grant_type: str = "client_credentials"
    scope: str


class AthenaTokenResponse(BaseModel):
    """OAuth2 token response."""

    model_config = ConfigDict(extra="allow")

    access_token: str
    token_type: str
    expires_in: int


class AthenaEncounterSummaryRequest(BaseModel):
    """GET /v1/{practice_id}/chart/encounters/{encounter_id}/summary"""

    model_config = ConfigDict(extra="allow")

    practice_id: str
    encounter_id: str


class AthenaEncounterSummaryResponse(BaseModel):
    """Encounter summary response."""

    model_config = ConfigDict(extra="allow")

    summaryhtml: str


class AthenaClinicalDocumentMeta(BaseModel):
    """Single document metadata item from the clinical documents list."""

    model_config = ConfigDict(extra="allow")

    clinicaldocumentid: int
    patientid: int
    documentdescription: str
    documentclass: str
    status: str
    internalnote: str | None = None
    createddatetime: str | None = None
    documentsource: str | None = None
    documentroute: str | None = None
    priority: str | None = None
    assignedto: str | None = None
    lastmodifieddatetime: str | None = None
    departmentid: str | None = None
    createddate: str | None = None
    lastmodifieduser: str | None = None
    lastmodifieddate: str | None = None
    observationdate: str | None = None
    createduser: str | None = None


class AthenaClinicalDocumentListRequest(BaseModel):
    """GET /v1/{practice_id}/patients/{patient_id}/documents/clinicaldocument"""

    model_config = ConfigDict(extra="allow")

    practice_id: str
    patient_id: str
    department_id: str | None = None
    limit: int = 200
    offset: int = 0


class AthenaClinicalDocumentListResponse(BaseModel):
    """Clinical documents list response."""

    model_config = ConfigDict(extra="allow")

    clinicaldocuments: list[AthenaClinicalDocumentMeta]
    totalcount: int | None = None


class AthenaClinicalDocumentContentRequest(BaseModel):
    """GET /v1/{practice_id}/patients/{patient_id}/documents/clinicaldocument/{document_id}"""

    model_config = ConfigDict(extra="allow")

    practice_id: str
    patient_id: str
    document_id: str


class AthenaClinicalDocumentContentResponse(BaseModel):
    """Clinical document content response."""

    model_config = ConfigDict(extra="allow")

    documentdata: str | None = None
    pages: list[dict] | None = None


class AthenaPushDocumentRequest(BaseModel):
    """POST /v1/{practice_id}/patients/{patient_id}/documents/clinicaldocument (future/deferred)"""

    model_config = ConfigDict(extra="allow")

    practice_id: str
    patient_id: str
    department_id: str
    attachment_contents: str
    attachment_type: str
    document_subclass: str = "JUNO_SUMMARY"
    internal_note: str


class AthenaPushDocumentResponse(BaseModel):
    """Push document response (future/deferred)."""

    model_config = ConfigDict(extra="allow")

    clinicaldocumentid: int
    success: bool | None = None

File: frontend/src/types/athena.ts
// frontend/src/types/athena.ts
// TypeScript interfaces mirroring the Athena Health API request/response models.
// These are the contract between Juno and Athena Health.
// Any change to Athena's API response shape must be reflected here first.

// Auth
export interface AthenaTokenResponse {
  access_token: string;
  token_type: string;
  expires_in: number;
  [key: string]: unknown;
}

// Encounters
export interface AthenaEncounterSummaryResponse {
  summaryhtml: string;
  [key: string]: unknown;
}

// Clinical Document metadata (from list endpoint)
export interface AthenaClinicalDocumentMeta {
  clinicaldocumentid: number;
  patientid: number;
  documentdescription: string;
  documentclass: string;
  status: string;
  internalnote?: string;
  createddatetime?: string;
  documentsource?: string;
  documentroute?: string;
  priority?: string;
  assignedto?: string;
  lastmodifieddatetime?: string;
  departmentid?: string;
  createddate?: string;
  lastmodifieduser?: string;
  lastmodifieddate?: string;
  observationdate?: string;
  createduser?: string;
  [key: string]: unknown;
}

export interface AthenaClinicalDocumentListResponse {
  clinicaldocuments: AthenaClinicalDocumentMeta[];
  totalcount?: number;
  [key: string]: unknown;
}

// Clinical Document content
export interface AthenaClinicalDocumentContentResponse {
  documentdata?: string;
  pages?: Array<{ pageid?: number; [key: string]: unknown }>;
  [key: string]: unknown;
}

// Push back (future/deferred)
export interface AthenaPushDocumentResponse {
  clinicaldocumentid: number;
  success?: boolean;
  [key: string]: unknown;
}

// ── Manifest models (static config files, not live API responses) ──────────

export interface AthenaEncounterManifestEntry {
  id: string;
  label: string;
  practice_id: string;
  patient_id: string;
  encounter_id: string;
  api_path: string;
  is_preview: boolean;
  preview_content: string | null;
}

export interface AthenaEncounterManifest {
  source_kind: "athena_encounter";
  label: string;
  tab_id: string;
  preview_entry_id: string;
  entries: AthenaEncounterManifestEntry[];
}

export interface AthenaClinicalDocManifestEntry {
  id: string;
  label: string;
  practice_id: string;
  patient_id: string;
  document_id: string;
  api_path: string;
  is_preview: boolean;
  preview_content: string | null;
}

export interface AthenaClinicalDocManifest {
  source_kind: "athena_clinical_doc";
  label: string;
  tab_id: string;
  preview_entry_id: string;
  entries: AthenaClinicalDocManifestEntry[];
}

export type AthenaManifest = AthenaEncounterManifest | AthenaClinicalDocManifest;
